### PR TITLE
updates dolt log to use sql backend

### PR DESCRIPTION
**Pull request by stephkyou**
Updates `dolt log` to use the appropriate sql engine to generate results.

Related: https://github.com/dolthub/dolt/issues/3922

### CODE REVIEW

File: go/cmd/dolt/commands/log.go [hunk 1]
```diff
@@ -283,56 +296,67 @@ func getCommitSpec(commit string) (*doltdb.CommitSpec, error) {
...
 		tagName = fmt.Sprintf("\033[33;1mtag: %s\033[0m", tagName) // tags names are bright yellow (33;1m)
-		cHashToRefs[t.Hash] = append(cHashToRefs[t.Hash], tagName)
+		cHashToRefs[hash.Parse(t[1].(string))] = append(cHashToRefs[hash.Parse(t[1].(string))], tagName)
 	}
 	return cHashToRefs, nil
 }
 
-func logCommits(ctx context.Context, dEnv *env.DoltEnv, opts *logOpts) int {
-	hashes := make([]hash.Hash, len(opts.commitSpecs))
+func logCommits(ctx context.Context, dEnv *env.DoltEnv, opts *logOpts, sqlResult []sql.Row, queryist cli.Queryist, sqlCtx *sql.Context) int {
```
Comment (macneale4) [inline go/cmd/dolt/commands/log.go:359]: Drop the dEnv since you don't want it. Keeps you honest :-)

File: go/cmd/dolt/commands/log.go [hunk 2]
```diff
@@ -399,34 +424,59 @@ func logCommits(ctx context.Context, dEnv *env.DoltEnv, opts *logOpts) int {
...
+			Description:   row[4].(string),
+			UserTimestamp: 0,
+		}
+		/*meta, mErr := comm.GetCommitMeta(ctx)
 		if mErr != nil {
 			cli.PrintErrln("error: failed to get commit metadata")
 			return 1
+		}*/
+
+		parents, err := GetRowsForSql(queryist, sqlCtx, fmt.Sprintf("select parent_hash from dolt_commit_ancestors where commit_hash = %s", row[0]))
```
Comment (macneale4) [inline go/cmd/dolt/commands/log.go:449]: ready row[0] into a variable above. Generally when you have an array at the top of the block and you have static indexes for for particular values, you want to copy them all into named values right away. When I read the lines 435 - 438, my first thought was 'where is row[0]`

File: go/cmd/dolt/commands/log.go [hunk 3]
```diff
@@ -399,34 +424,59 @@ func logCommits(ctx context.Context, dEnv *env.DoltEnv, opts *logOpts) int {
...
+		}*/
+		cmHash := hash.Parse(row[0].(string))
 
 		commitsInfo = append(commitsInfo, logNode{
 			commitMeta:   meta,
 			commitHash:   cmHash,
 			parentHashes: pHashes,
 			branchNames:  cHashToRefs[cmHash],
-			isHead:       cmHash == *cwbHash})
+			isHead:       row[0] == headHash})
```
Comment (macneale4) [inline go/cmd/dolt/commands/log.go:480]: double down on my suggestion above

File: go/cmd/dolt/commands/log.go [hunk 4]
```diff
@@ -131,8 +154,87 @@ func (cmd LogCmd) logWithLoggerFunc(ctx context.Context, commandStr string, args
...
+// constructInterpolatedDoltLogQuery generates the sql query necessary to call the DOLT_LOG() function.
+// Also interpolates this query to prevent sql injection.
+func constructInterpolatedDoltLogQuery(apr *argparser.ArgParseResults) (string, error) {
+	var params []interface{}
+
+	var buffer bytes.Buffer
+	var first bool
+	first = true
+
+	if apr.Contains(cli.ParentsFlag) && apr.Contains(cli.DecorateFlag) {
```
Comment (macneale4) [inline go/cmd/dolt/commands/log.go:170]: You can simplify this. For example, just alway send the --decorate flag, and always expect that column to exist. There are other examples of this too. Eg, if they don't specify a max number, then we should supply one.

**Comment by stephkyou** [comment]
Closing to re-work `dolt_log` table function

**Pull request closed without merging**